examples/gno.land/r/gov/dao/v3/impl/govdao.gno

```diff
@@ -0,0 +1,241 @@
+package impl
+
+import (
+	"errors"
+	"std"
+	"strconv"
+
+	"gno.land/p/demo/avl/pager"
+	"gno.land/p/demo/mux"
+	"gno.land/p/demo/ufmt"
+	"gno.land/r/gov/dao"
+	"gno.land/r/gov/dao/v3/memberstore"
+)
+
+var ErrMemberNotFound = errors.New("member not found")
+
+type GovDAO struct {
+	pss      ProposalsStatuses
+	pssPager *pager.Pager
+	router   *mux.Router
+}
+
+func NewGovDAO() *GovDAO {
+	pss := NewProposalsStatuses()
+	d := &GovDAO{
+		pss:      pss,
+		pssPager: pager.NewPager(pss.Tree, 3, true),
+	}
+
+	r := mux.NewRouter()
+
+	r.HandleFunc("/", func(rw *mux.ResponseWriter, req *mux.Request) {
+		rw.Write(d.renderActiveProposals(req.RawPath))
+	})
+
+	r.HandleFunc("", func(rw *mux.ResponseWriter, req *mux.Request) {
+		rw.Write(d.renderActiveProposals(req.RawPath))
+	})
+
+	r.HandleFunc("{pid}", func(rw *mux.ResponseWriter, req *mux.Request) {
+		rw.Write(d.renderProposal(req.GetVar("pid")))
+	})
+
+	r.HandleFunc("{pid}/votes", func(rw *mux.ResponseWriter, req *mux.Request) {
+		rw.Write(d.renderVotesForProposal(req.GetVar("pid")))
+	})
+
+	d.router = r
+	_govdao = d
+
+	return d
+}
+
+// Setting this to a global variable forces attaching the GovDAO struct to this realm.
+var _govdao *GovDAO
+
+func (g *GovDAO) PreCreateProposal(r dao.ProposalRequest) (dao.Author, error) {
+	if !g.isValidCall() {
+		return nil, errors.New("proposal creation must be done directly by a user")
+	}
+
+	return NewUserAuthor(), nil
+}
+
+func (g *GovDAO) PostCreateProposal(r dao.ProposalRequest, pid dao.ProposalID) {
+	// Tiers Allowed to Vote
+	tatv := []string{memberstore.T1, memberstore.T2, memberstore.T3}
+	switch v := r.Metadata().(type) {
+	case *AddMemberMetadata:
```

Comment:
This is difficult to maintain and compose. An interface like `type T1T2Only interface { IsT1T2Only() }` seems more appropriate. However, as I suggested elsewhere, the entire concept of Metadata is too complex and intertwined in many areas. We should simplify it by creating "ProposalFilters" that can be made generic and reused, while keeping callbacks straightforward and allowing descriptions to provide the necessary clarification.